source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -31,37 +31,55 @@ using ::testing::SetArgPointee;
 using ::testing::SetArrayArgument;
 using ::testing::Throw;
 
+using FakeResourceRepository = nidevice_grpc::SessionResourceRepository<ViSession>;
+
 const std::uint32_t kTestViSession = 12345678;
 const std::uint32_t kDriverSuccess = 0;
 const std::uint32_t kDriverFailure = 1;
 const char* kTestChannelName = "channel";
 
-std::uint32_t create_session(nidevice_grpc::SessionRepository& session_repo, ViSession sessionToStore)
+nifake_grpc::NiFakeService create_fake_service(NiFakeMockLibrary* library, nidevice_grpc::SessionRepository* session_repository)
 {
-  auto init_lambda = [&]() -> std::tuple<int, uint32_t> {
-    return std::make_tuple(0, sessionToStore);
-  };
-  uint32_t session_id;
-  session_repo.add_session("", init_lambda, NULL, session_id);
-  return session_id;
+  auto resource_repository = std::make_shared<FakeResourceRepository>(session_repository);
+  return nifake_grpc::NiFakeService(library, resource_repository);
 }
 
-std::uint32_t create_session(std::string session_name, nidevice_grpc::SessionRepository& session_repo, ViSession sessionToStore)
+std::int32_t create_session(
```

Comment:
Note: In an earlier version of this change, I had the services each own their own `SessionResourceRepository`. Then I discovered that this broke `TClk` and reverted to taking in the `SessionResourceRepository` as a `shared_ptr` (technically this could be conditional on whether-or-not the resources are share-able, but the codegen is simpler if it's always shared).

In that earlier version of the change, I removed the "seam" that made it possible to inject sessions directly. So I updated this test helper to add sessions through the init method. When I switch to a `shared_ptr` I reintroduced a similar seam. But, I like the tests like this. I think it's a more consistent "level" for this test class. And I think it's an advantage that it "works" with different implementations.